### PR TITLE
Update definitions for react-dropzone

### PR DESCRIPTION
**Pull request by LKay**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/react-dropzone/index.d.ts
Comment (Kovensky) [inline types/react-dropzone/index.d.ts]: This either makes the first argument's inferred type `E | ImageFile[]` or makes it non-inferrable 🤔

Comment (Kovensky) [inline types/react-dropzone/index.d.ts]: Reading into the source, it seems this got confused between `this.onDrop` and `this.props.onDrop`. `this.onDrop` is the `DragEvent` handler, and is internal to dropzone;  the handler given on props is `(accepted: ImageFile[], rejected: ImageFile[], event: React.DragEvent<HTMLDivElement>>): void`.

It could also include the `this` type (`(this: Dropzone, ...`) but not sure it's a good idea.

File: types/react-dropzone/index.d.ts
Comment (Kovensky) [inline types/react-dropzone/index.d.ts]: `onDropAccepted` and `onDropRejected` are missing the _second_ argument, so you can't just use an overload. Need to make a separate type that is something like `(acceptedOrRejected: ImageFile[], event: React.DropEvent<HTMLDivElement>): void`.

File: types/react-dropzone/index.d.ts
Comment (Kovensky) [inline types/react-dropzone/index.d.ts]: https://github.com/okonet/react-dropzone/blob/master/src/index.js#L533 This one is a default export 😄

Comment (Kovensky) [inline types/react-dropzone/index.d.ts]: ...nevermind, they use `add-module-exports` 🤦‍♂️

File: types/react-dropzone/index.d.ts
Comment (benbayard) [inline types/react-dropzone/index.d.ts]: Can you point me to the source where the function is called with this? I cannot find it.

File: types/react-dropzone/index.d.ts
Comment (benbayard) [inline types/react-dropzone/index.d.ts]: Yup, it's really unfortunate. :(

File: types/react-dropzone/react-dropzone-tests.tsx
Comment (benbayard) [inline types/react-dropzone/react-dropzone-tests.tsx]: extra lines?

File: types/react-dropzone/v2/react-dropzone-tests.tsx
Comment (benbayard) [inline types/react-dropzone/v2/react-dropzone-tests.tsx]: not needed.

File: types/react-dropzone/v2/react-dropzone-tests.tsx
Comment (benbayard) [inline types/react-dropzone/v2/react-dropzone-tests.tsx]: I would declare a function and pass it in here for easier maintenance in the future.

File: types/react-dropzone/react-dropzone-tests.tsx
Comment (benbayard) [inline types/react-dropzone/react-dropzone-tests.tsx]: use the `require` syntax as this is a `export =` (commonjs) module.

File: types/react-dropzone/react-dropzone-tests.tsx
Comment (benbayard) [inline types/react-dropzone/react-dropzone-tests.tsx]: keep the spacing around values consistent ( the v2 typings do not have spaces ). i'd keep the original

File: types/react-dropzone/index.d.ts
Comment (LKay) [inline types/react-dropzone/index.d.ts]: This is a hack that allows override type default handler for `onDrop`. This is a standard js handlers which is defined in `DOMAttributes` through `InputHTMLAttributes`.

File: types/react-dropzone/v2/react-dropzone-tests.tsx
Comment (LKay) [inline types/react-dropzone/v2/react-dropzone-tests.tsx]: Same as above.

File: types/react-dropzone/index.d.ts
Comment (Kovensky) [inline types/react-dropzone/index.d.ts]:
```ts
type DropFilesEventHandler = React.DragEventHandler<HTMLDivElement> & ((accepted: ImageFile[], rejected: imageFile[], event: DragEvent<HTMLDivElement>) => void)
type DropFileAcceptedOrRejectedEventHandler = (acceptedOrRejected: ImageFile[], event: DragEvent<HTMLDivElement>) => void
```

Comment (LKay) [inline types/react-dropzone/index.d.ts]: Well, I did _exactly_ that with `Pick`. ;)

**Review by Jessidhia — APPROVED** [review]

**Comment by RyanCavanaugh** [comment]
Bad bot, don't ping people who signed off...

File: types/react-dropzone/react-dropzone-tests.tsx
Comment (benbayard) [inline types/react-dropzone/react-dropzone-tests.tsx]: rm space

File: types/react-dropzone/v2/react-dropzone-tests.tsx
Comment (benbayard) [inline types/react-dropzone/v2/react-dropzone-tests.tsx]: this appears to be out of sync w/ v1. (updates were made there that are not here)

**Review by benbayard — APPROVED** [review]
Overall this LGTM.

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR appears ready to merge pending express review by a maintainer.

**Pull request merged**
